src/main/java/com/o19s/es/ltr/action/TransportFeatureStoreAction.java

```diff
@@ -121,7 +113,7 @@ private IndexRequest buildIndexRequest(Task parentTask, FeatureStoreRequest requ
         return indexRequest;
     }
 
-    private void validate(FeatureStoreRequest request) {
+    private void validateAndSave(FeatureStoreRequest request, Task task, ActionListener<FeatureStoreResponse> listener) {
```

Comment:
Looking at the code, this seems to be `validateOrSave()`. I think this would all be more readable by splitting in 2 methods (`validate` / `save`) and move the `if (request.getValidation() != null` to `doExecute()`. But I might be missing some dependency...